source/core/DualTypeVisitor.cpp

```diff
@@ -188,19 +178,23 @@ namespace Vireo
     }
 
     //------------------------------------------------------------
-    bool DualTypeVisitor::ArrayCompatible(TypeRef typeRefX, TypeRef typeRefY, DualTypeOperation* operation)
+    bool DualTypeVisitor::ArrayCompatible(TypeRef typeRefX, void* pDataX, TypeRef typeRefY, void* pDataY, DualTypeOperation* operation)
     {
         // Verify number of dimensions are the same
         bool success = typeRefX->Rank() == typeRefY->Rank();
         // Verify each dimension has the same size
         if (success) {
-            TypedArrayCoreRef arrayX = *(static_cast<const TypedArrayCoreRef*>(typeRefX->Begin(kPARead)));
-            TypedArrayCoreRef arrayY = *(static_cast<const TypedArrayCoreRef*>(typeRefY->Begin(kPARead)));
+            TypedArrayCoreRef arrayX = *(static_cast<const TypedArrayCoreRef*>(pDataX));
+            TypedArrayCoreRef arrayY = *(static_cast<const TypedArrayCoreRef*>(pDataY));
+            if (operation->ShouldInflateDestination()) {
```

Comment:
This is an important change. I had to allow the operation to decide if the target needs to be inflated before they're considered compatible. Since there is not real data associated with the output terminal (yet), we do not want them to be considered incompatible.